docs/enhancements/easy-discovery-password.md

```diff
@@ -0,0 +1,174 @@
+---
+title: easy-discovery-password
+authors:
+  - "@omertuc"
+creation-date: 2022-08-02
+last-updated: 2022-08-02
+---
+
+# Easy discovery login password
+
+Give users an easy, safe way to set the discovery ISO ignition
+
+## Summary
+
+We want to improve the UI to allow the user to set the `core` user password for
+the discovery ISO so they can easily login to their cluster and troublehshoot
+networking issues, in the absence of a working SSH connection to the server.
+
+In addition, we should improve the API and documnetation to make it possible
+for users to set an ignition `core` user password without having to deal with
+complicated ignition overrides.
+
+## Motivation
+
+The discovery ISO generated by the Assisted Installer comes with a `core` user
+that has its password login disabled (although SSH access using the provided
+public key is still supported).
+
+However, it can sometimes be useful (e.g. when the network is misbehaving, SSH
+isn't working, and you want to troubleshoot why), to login to the machine
+directly (with physical or through server remote management solutions), using a
+password. This can be done using the console login prompt that is shown after
+RHCOS is done booting.
+
+To allow that, the user must have a password set. In order to set a password,
+it's possible to use the Assisted Installer ignition override API which
+provides support for editing the discovery ISO's ignition file with custom
+parameters.
+
+This method is complicated and should be made easier.
+
+### Goals
+
+- Add new UI to set the `core` user password for the discovery ISO 
+- Add new API to set the ignition password hash for the discovery ISO `core`
+user password, with documentation explaining how it can be done
+- Do all of that in a secure way. At no point should the service know the
+user's password. But it's okay if the service knows the hash, as that doesn't
+reveal the password
+
+### Non-Goals
+
+- Allow the core user password to be set even after the host has been
+discovered. We could technically create a new agent step that would set the
+discovery password hash to whatever the user wants on a live system, but that
+would be useless in 99% of cases since if the networking is working well enough
+for the agent to connect to the service, it's probably working well enough for
+the user to SSH into the server using their public key anyway - no password
+needed.
+
+## Proposal
+
+### New service APIs
+
+#### REST API
+
+- Add a new optional `discovery_core_user_password_hash` parameter to the
+`infra-env-create-params` API structure. Ensure this parameter has
```

Comment:
"discovery" is redundant because it's already part of the infra-env.  "core_user_password_hash" is sufficient.

Reply:
It should also be in infra-env-update-params and infra-env, otherwise it sounds like it can be set on create and never updated or viewed.